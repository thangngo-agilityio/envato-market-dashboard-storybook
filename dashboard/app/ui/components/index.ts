export { default as InputField } from './common/InputField';
export { default as FallbackErrorBoundary } from './FallbackErrorBoundary';
export { default as Navigation } from './common/Navigation';
export { default as Menu } from './common/Menu';
export { default as Logo } from './common/Logo';
export { default as ExpandSidebar } from './ExpandSidebar';
export { default as MiniSidebar } from './MiniSidebar';
export * from './Icons';
export * from './Images';
export { default as Indicator } from './Indicator';
export { default as Loading } from './common/Loading';
export { default as Avatar } from './common/Avatar';
export { default as Benefit } from './common/Benefit';
export { default as Divider } from './common/Divider';
export { default as IconButton } from './common/IconButton';
export { default as Dropdown } from './common/Dropdown';
export { default as Select } from './common/Select';
export { default as Pagination } from './common/Pagination';
export { default as Button } from './common/Button';
export { default as Table } from './common/Table';
export { default as CustomerNameCell } from './common/Table/Body/CustomerNameCell';
export { default as ProductNameCell } from './common/Table/Body/ProductNameCell';
export { default as ActionCell } from './common/Table/Body/ActionCell';
export { default as UserInfoCell } from './UsersTable/Cell/UserInfoCell';
export { default as StatusCell } from './UsersTable/Cell/StatusCell';
export { default as HeadCell } from './common/Table/HeadCell';
export { default as Selector } from './TransactionTable/Selector';
export { default as TransactionTable } from './TransactionTable';
export { default as SearchBar } from './common/SearchBar';
export { default as SwitchTheme } from './SwitchTheme';
export { default as CardPayment } from './CardPayment';
export { default as BoxChat } from './BoxChat';
export { default as TotalStatisticCard } from './TotalStatisticCard';
export { default as Fetching } from './Skeleton/Fetching';
export { default as TotalStatisticListSkeleton } from './Skeleton/TotalStatisticListSkeleton';
export { default as TotalStatisticList } from './TotalStatisticList';
export { default as RevenueFlow } from './RevenueFlow';
export { default as Efficiency } from './Efficiency';
export { default as TotalBalance } from './TotalBalance';
export { default as OverallBalance } from './OverallBalance';
export { default as UserCard } from './UserCard';
export { default as ItemSideBarSetting } from './ItemSideBarSetting';
export { default as FaqItem } from './Setting/Faq/FaqItem';
export { default as UpdateProfile } from './Profile';
export { default as AuthForm } from './AuthForm';
export { default as Modal } from './common/Modal';
export { default as Notification } from './common/Notification';
export { default as FallbackImage } from './FallbackImage';
export { default as ChatMember } from './ChatMember';
export { default as Conversation } from './Conversation';
export { default as BonusNotification } from './common/BonusNotification';
export { default as CardIssues } from './CustomerIssues/CardIssues';
export { default as CustomerIssues } from './CustomerIssues';
export { default as PinCode } from './common/PinCode';
export { default as Quill } from './Quill';
export { default as MessageAdmin } from './MessagesAdmin';
export { default as ListMessages } from './Conversation/ListMessage';
export { default as InputSendMessages } from './InputSendMessages';
export { default as PinCodeModal } from './PinCodeModal';
export { default as ProductsTable } from './ProductsTable';
export { default as UploadImages } from './UploadImages';
export { default as ProductForm } from './common/Table/Body/ProductForm';
export { default as RecentActivitiesTable } from './RecentActivitiesTable';
