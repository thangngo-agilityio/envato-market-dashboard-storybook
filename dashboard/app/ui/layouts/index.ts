export { default as SideBar } from './Sidebar';
export { default as Header } from './Header';
export { default as AuthHeader } from './AuthHeader';
export { default as AuthFooter } from './AuthFooter';
export { default as MainLayout } from './MainLayout';
