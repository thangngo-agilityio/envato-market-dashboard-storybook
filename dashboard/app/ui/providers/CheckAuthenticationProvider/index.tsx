'use client';

import { PRIVATE_ROUTES, PUBLIC_ROUTES, ROUTES } from '@/lib/constants';
import { TAuthStoreData, authStore } from '@/lib/stores';
import { redirect, usePathname } from 'next/navigation';
import { useEffect } from 'react';

type TValidateRoute = {
  id: number;
  path: string;
};

const CheckAuthenticationProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const pathname = usePathname();
  const user = authStore((state): TAuthStoreData['user'] => state.user);
  const isMatchPrivateRoute: boolean = PRIVATE_ROUTES.some(
    (route: TValidateRoute) =>
      pathname === ROUTES.ROOT || `/${route.path}` === pathname,
  );
  const isMatchPublicRoute: boolean = PUBLIC_ROUTES.some(
    (route: TValidateRoute) => `/${route.path}` === pathname,
  );

  useEffect(() => {
    if (isMatchPublicRoute && !!user) {
      return redirect(ROUTES.ROOT);
    }

    if (isMatchPrivateRoute && !user) {
      return redirect(ROUTES.LOGIN);
    }
  }, [isMatchPrivateRoute, isMatchPublicRoute, user]);

  return children;
};

export default CheckAuthenticationProvider;
