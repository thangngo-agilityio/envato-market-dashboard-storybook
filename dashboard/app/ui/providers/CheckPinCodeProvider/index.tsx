'use client';

import dynamic from 'next/dynamic';
import { useCallback, useMemo } from 'react';
import { useDisclosure, useToast } from '@chakra-ui/react';

import { SubmitHandler, useForm } from 'react-hook-form';

//Constants
import { ERROR_MESSAGES, STATUS, SUCCESS_MESSAGES } from '@/lib/constants';

// Hooks
import { useAuth, usePinCode } from '@/lib/hooks';

// Types
import { TPinCodeForm } from '@/lib/interfaces';

// Stores
import { TAuthStoreData, authStore } from '@/lib/stores';

// Utils
import { customToast } from '@/lib/utils';

const Modal = dynamic(() => import('@/ui/components/common/Modal'));
const PinCode = dynamic(() => import('@/ui/components/common/PinCode'));

const CheckPinCodeProvider = () => {
  const user = authStore((state): TAuthStoreData['user'] => state.user);

  const { isOpen: isPinCodeModalOpen, onClose: onClosePinCodeModal } =
    useDisclosure({ isOpen: true });

  const {
    control,
    handleSubmit,
    formState: { isValid, isSubmitting },
  } = useForm<TPinCodeForm>({
    defaultValues: {
      pinCode: '',
    },
  });

  const { setUser } = useAuth();

  const toast = useToast();

  const { handleSetPinCode } = usePinCode();

  const onSubmitPinCode: SubmitHandler<TPinCodeForm> = useCallback(
    (data) => {
      if (user) {
        data.userId = user.id;

        try {
          handleSetPinCode(data);

          setUser({ user: { ...user, pinCode: data.pinCode } });

          onClosePinCodeModal();

          toast(
            customToast(
              SUCCESS_MESSAGES.SET_PIN_CODE.title,
              SUCCESS_MESSAGES.SET_PIN_CODE.description,
              STATUS.SUCCESS,
            ),
          );
        } catch (error) {
          toast(
            customToast(
              ERROR_MESSAGES.SET_PIN_CODE.title,
              ERROR_MESSAGES.SET_PIN_CODE.description,
              STATUS.ERROR,
            ),
          );
        }
      }
    },
    [handleSetPinCode, onClosePinCodeModal, setUser, toast, user],
  );

  const pinCodeModalBody = useMemo(
    () => (
      <PinCode
        control={control}
        isDisabled={!isValid || isSubmitting}
        onSubmit={handleSubmit(onSubmitPinCode)}
        onClose={onClosePinCodeModal}
      />
    ),
    [
      control,
      handleSubmit,
      isSubmitting,
      isValid,
      onClosePinCodeModal,
      onSubmitPinCode,
    ],
  );

  return (
    isPinCodeModalOpen && (
      <Modal
        title="Please set the PIN code to your account"
        isOpen={isPinCodeModalOpen}
        onClose={onClosePinCodeModal}
        body={pinCodeModalBody}
      />
    )
  );
};

export default CheckPinCodeProvider;
