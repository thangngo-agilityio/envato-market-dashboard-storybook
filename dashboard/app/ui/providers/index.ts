export { default as ChakraProvider } from './ChakraUIProvider';
export { default as QueryProvider } from './QueryProvider';
export { default as CheckAuthenticationProvider } from './CheckAuthenticationProvider';
export { default as CheckPinCodeProvider } from './CheckPinCodeProvider';
