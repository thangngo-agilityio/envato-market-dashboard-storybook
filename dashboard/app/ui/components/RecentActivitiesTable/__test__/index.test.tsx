import { useSearchParams, useRouter } from 'next/navigation';
import axios from 'axios';

// Components
import { RecentActivitiesTable } from '../..';

// Hooks
import * as hooks from '@/lib/hooks';

// Provider
import { QueryProvider } from '@/ui/providers';

// Mocks
import { RECENT_ACTIVITIES } from '@/lib/mocks';

// Constants
import { DEBOUNCE_TIME } from '@/lib/constants';

jest.mock('@/lib/hooks', () => ({
  ...jest.requireActual('@/lib/hooks'),
  usePagination: jest.fn(),
  useRecentActivities: jest.fn(),
}));

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useToast: () => jest.fn(),
}));

jest.mock('axios');

const resetMock = jest.fn();
const sortByMock = jest.fn();

const SORT_ICON = 'sort-icon';
const SEARCH_ACTIVITY = 'search-transaction';

const setup = () =>
  render(<RecentActivitiesTable />, {
    wrapper: QueryProvider,
  });

describe('Recent Activities table', () => {
  beforeEach(() => {
    jest.spyOn(hooks, 'usePagination').mockReturnValue({
      resetPage: resetMock,
      filterData: RECENT_ACTIVITIES,
      data: [1],
    } as unknown as ReturnType<typeof hooks.usePagination>);
    jest.spyOn(axios, 'get').mockResolvedValue(RECENT_ACTIVITIES);
    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn(),
      forEach: jest.fn(),
    });
    (useRouter as jest.Mock).mockReturnValue({
      push: jest.fn(),
    });
    jest.spyOn(hooks, 'useRecentActivities').mockReturnValue({
      activities: RECENT_ACTIVITIES,
      sortBy: sortByMock,
      isLoading: false,
    } as unknown as ReturnType<typeof hooks.useRecentActivities>);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Match with snapshot with default prop', () => {
    const { container } = setup();

    waitFor(() => expect(container).toMatchSnapshot());
  });

  jest.useFakeTimers();
  it('Typing input search', async () => {
    const { getByTestId } = setup();
    const input = getByTestId(SEARCH_ACTIVITY) as HTMLInputElement;

    await act(async () => {
      fireEvent.change(input, {
        target: {
          value: 'update',
        },
      });
    });

    jest.advanceTimersByTime(DEBOUNCE_TIME);
    expect(resetMock).toHaveBeenCalled();
  });

  it('Sort action', async () => {
    const { getAllByTestId } = setup();
    const sortIcon = getAllByTestId(SORT_ICON);

    await act(async () => {
      fireEvent.click(sortIcon[0]);
    });

    expect(sortByMock).toHaveBeenCalled();
  });
});
