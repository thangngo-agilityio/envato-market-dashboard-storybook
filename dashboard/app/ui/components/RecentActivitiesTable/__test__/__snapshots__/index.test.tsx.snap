// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent Activities table Match with snapshot with default prop 1`] = `
<div>
  <div
    class="css-194coc3"
  >
    <form
      class="chakra-stack css-16wym0o"
      data-testid="search-bar"
    >
      <div
        class="css-1b9rej7"
      >
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <div
            class="chakra-input__group css-1091msq"
            data-group="true"
          >
            <div
              class="chakra-input__left-element css-1kg931p"
            >
              <svg
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              class="chakra-input css-1ybypo"
              data-testid="search-transaction"
              id="field-:r0:"
              placeholder="Search by name or email"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="css-tzs3vb"
      >
        <button
          aria-controls="menu-list-:r2:"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-button chakra-menu__menu-button css-1d0mwls"
          id="menu-button-:r2:"
          type="button"
        >
          <span
            class="css-xl71ch"
          >
            <span
              class="css-ccj5l2"
            >
              <svg
                fill="none"
                height="17"
                stroke="#1D1E24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.552 13.502H1.25M10.362 3.81h6.301"
                />
                <path
                  clip-rule="evenodd"
                  d="M5.948 3.756a2.355 2.355 0 0 0-2.363-2.347 2.355 2.355 0 0 0-2.363 2.347 2.355 2.355 0 0 0 2.363 2.346 2.355 2.355 0 0 0 2.363-2.346ZM17.221 13.463a2.354 2.354 0 0 0-2.362-2.346 2.355 2.355 0 0 0-2.364 2.346 2.355 2.355 0 0 0 2.364 2.347 2.354 2.354 0 0 0 2.362-2.347Z"
                />
              </svg>
              Filters
            </span>
          </span>
        </button>
        <div
          class="css-r6z5ec"
          style="visibility: hidden; position: absolute; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-1k4isgz"
            id="menu-list-:r2:"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="0"
              id="menu-list-:r2:-menuitem-:r3:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Default
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="1"
              id="menu-list-:r2:-menuitem-:r4:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Sign in
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="2"
              id="menu-list-:r2:-menuitem-:r5:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Sign out
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="3"
              id="menu-list-:r2:-menuitem-:r6:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Sign up
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="4"
              id="menu-list-:r2:-menuitem-:r7:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Forgot password
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="5"
              id="menu-list-:r2:-menuitem-:r8:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Add money
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="6"
              id="menu-list-:r2:-menuitem-:r9:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Send money
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="7"
              id="menu-list-:r2:-menuitem-:ra:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Update transaction
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="8"
              id="menu-list-:r2:-menuitem-:rb:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Delete transaction
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="9"
              id="menu-list-:r2:-menuitem-:rc:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Create product
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="10"
              id="menu-list-:r2:-menuitem-:rd:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Update product
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="11"
              id="menu-list-:r2:-menuitem-:re:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Delete product
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="12"
              id="menu-list-:r2:-menuitem-:rf:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Create issues
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="13"
              id="menu-list-:r2:-menuitem-:rg:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Save profile
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-o9weuj"
              data-index="14"
              id="menu-list-:r2:-menuitem-:rh:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-15asmb0"
              >
                Save change password
              </p>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    class="css-17qnj1m"
  >
    <div
      class="chakra-table__container css-1d9a4hd"
    >
      <table
        class="chakra-table css-5605sr"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="ID Action"
                >
                  ID Action
                </p>
                <button
                  aria-label="This is the icon for ID Action"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Action Name"
                >
                  Action Name
                </p>
                <button
                  aria-label="This is the icon for Action Name"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Email"
                >
                  Email
                </p>
                <button
                  aria-label="This is the icon for Email"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Date"
                >
                  Date
                </p>
                <button
                  aria-label="This is the icon for Date"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
            data-testid="table-row"
          >
            <td
              class="css-1xp2ynx"
            >
              <div
                class="css-1h2sv0j"
              >
                <p
                  class="chakra-text css-19zvnok"
                >
                  1
                </p>
              </div>
            </td>
            <td
              class="css-1xp2ynx"
            >
              <div
                class="css-1h2sv0j"
              >
                <p
                  class="chakra-text css-107vws0"
                >
                  Name 1
                </p>
              </div>
            </td>
            <td
              class="css-5x3kvh"
            >
              <a
                class="chakra-text css-186o5zi"
                href="mailto:email 1"
              >
                email 1
              </a>
            </td>
            <td
              class="css-1v6q597"
            >
              <p
                class="chakra-text css-gmde4n"
              >
                00:00:00 Jan 01, 2001
              </p>
            </td>
          </tr>
          <tr
            class="css-0"
            data-testid="table-row"
          >
            <td
              class="css-1xp2ynx"
            >
              <div
                class="css-1h2sv0j"
              >
                <p
                  class="chakra-text css-19zvnok"
                >
                  2
                </p>
              </div>
            </td>
            <td
              class="css-1xp2ynx"
            >
              <div
                class="css-1h2sv0j"
              >
                <p
                  class="chakra-text css-107vws0"
                >
                  Name 2
                </p>
              </div>
            </td>
            <td
              class="css-5x3kvh"
            >
              <a
                class="chakra-text css-186o5zi"
                href="mailto:email 2"
              >
                email 2
              </a>
            </td>
            <td
              class="css-1v6q597"
            >
              <p
                class="chakra-text css-gmde4n"
              >
                00:00:00 Feb 01, 2001
              </p>
            </td>
          </tr>
          <tr
            class="css-0"
            data-testid="table-row"
          >
            <td
              class="css-1xp2ynx"
            >
              <div
                class="css-1h2sv0j"
              >
                <p
                  class="chakra-text css-19zvnok"
                >
                  3
                </p>
              </div>
            </td>
            <td
              class="css-1xp2ynx"
            >
              <div
                class="css-1h2sv0j"
              >
                <p
                  class="chakra-text css-107vws0"
                >
                  Name 3
                </p>
              </div>
            </td>
            <td
              class="css-5x3kvh"
            >
              <a
                class="chakra-text css-186o5zi"
                href="mailto:email 3"
              >
                email 3
              </a>
            </td>
            <td
              class="css-1v6q597"
            >
              <p
                class="chakra-text css-gmde4n"
              >
                00:00:00 Mar 01, 2001
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="css-2qrmgs"
  >
    <div
      class="css-1mngwok"
      data-testid="pagination"
    >
      <div
        class="css-1y90n2a"
      >
        <p
          class="chakra-text css-r5mmfv"
        >
          Show result:
        </p>
        <div
          class="css-110fpu2"
        >
          <button
            aria-controls="menu-list-:r15:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-button chakra-menu__menu-button css-1d0mwls"
            id="menu-button-:r15:"
            type="button"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-1l4w6pd"
              >
                <p
                  class="chakra-text css-1aitddj"
                >
                  10
                </p>
                <div
                  class="css-t4r95o"
                >
                  <svg
                    fill="none"
                    height="15"
                    stroke="#A0AEC0"
                    style="transform: rotate(undefined);"
                    viewBox="0 0 18 15"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 10L12 14L17 10"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </span>
          </button>
          <div
            class="css-r6z5ec"
            style="visibility: hidden; position: absolute; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-1k4isgz"
              id="menu-list-:r15:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-o9weuj"
                data-index="0"
                id="menu-list-:r15:-menuitem-:r16:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <p
                  class="chakra-text css-15asmb0"
                >
                  10
                </p>
              </button>
              <button
                class="chakra-menu__menuitem css-o9weuj"
                data-index="1"
                id="menu-list-:r15:-menuitem-:r17:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <p
                  class="chakra-text css-15asmb0"
                >
                  20
                </p>
              </button>
              <button
                class="chakra-menu__menuitem css-o9weuj"
                data-index="2"
                id="menu-list-:r15:-menuitem-:r18:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <p
                  class="chakra-text css-15asmb0"
                >
                  50
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-69i1ev"
      >
        <button
          aria-label="btn-prev"
          class="chakra-button css-1u5n2r4"
          data-testid="prev-button"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            stroke="#A0AEC0"
            style="transform: rotate(90deg);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10L12 14L17 10"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="css-70qvj9"
        />
        <button
          aria-label="btn-next"
          class="chakra-button css-5dzvs"
          data-testid="next-button"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            stroke="#A0AEC0"
            style="transform: rotate(-90deg);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10L12 14L17 10"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
